v2/blockstore/readonly_test.go

```diff
@@ -235,3 +235,24 @@ func newV1Reader(r io.Reader, zeroLenSectionAsEOF bool) (*carv1.CarReader, error
 	}
 	return carv1.NewCarReader(r)
 }
+
+func TestReadOnlyErrorAfterClose(t *testing.T) {
+	bs, err := OpenReadOnly("../testdata/sample-v1.car")
+	require.NoError(t, err)
+
+	roots, err := bs.Roots()
+	require.NoError(t, err)
+	_, err = bs.Get(roots[0])
+	require.NoError(t, err)
+	_, err = bs.GetSize(roots[0])
+	require.NoError(t, err)
+
+	bs.Close()
+
+	_, err = bs.Roots()
+	require.Error(t, err)
+	_, err = bs.Get(roots[0])
+	require.Error(t, err)
+	_, err = bs.GetSize(roots[0])
+	require.Error(t, err)
+}
```

Comment:
Would it make sense to add `AllKeysChan` while at it? 

A separate test-case for close with `AllKeysChan` mid iteration would also be good.

Reply:
> Would it make sense to add `AllKeysChan` while at it?

Alright, I've added all read-only methods.

> A separate test-case for close with `AllKeysChan` mid iteration would also be good.

I agree this would be nice, but we already lacked a test before, and I want to merge this sooner than later for ignite. I'll just add a TODO.